src/network/networkd-link.c

```diff
@@ -587,6 +591,61 @@ int link_get(Manager *m, int ifindex, Link **ret) {
         return 0;
 }
 
+int link_get_by_name(const Manager *m, const char *ifname, Link **ret) {
+        Link *link;
+
+        assert(m);
+        assert(ifname);
+        assert(ret);
+
+        link = hashmap_get(m->links_by_name, ifname);
+        if (!link)
+                return -ENODEV;
+
+        *ret = link;
+
+        return 0;
+}
+
+int link_update_by_name(Link *link) {
+        char **n;
+        int r;
+
+        assert(link);
+
+        r = hashmap_ensure_allocated(&link->manager->links_by_name, &string_hash_ops);
+        if (r < 0)
+                return r;
+
+        r = hashmap_put(link->manager->links_by_name, link->ifname, link);
+        if (r < 0)
+                return r;
+
+        STRV_FOREACH(n, link->alternative_names) {
+                r = hashmap_put(link->manager->links_by_name, *n, link);
+                if (r < 0)
+                        return r;
+        }
+
+        return 0;
+}
+
+int link_remove_by_name(Link *link) {
+        char **n;
+
+        assert(link);
+
+        if (hashmap_size(link->manager->links_by_name) <= 0)
+                return 0;
```

Comment:
this check looks unnecessar, should work the same if these two lines are dropped